src/main/java/pl/sknikod/kodemy/material/MaterialService.java

```diff
@@ -27,12 +40,32 @@ public MaterialCreateResponse create(MaterialCreateRequest body) {
                 .getOrElseThrow(() -> new GeneralRuntimeException("Failed to processing"));
     }
 
+    public void addGrade(MaterialAddGradeRequest body, Long materialId) {
+        Optional<Material> material = Option.of(materialRepository.findById(materialId))
+                .getOrElseThrow(() -> new GeneralRuntimeException("Failed to processing"));
```

Comment:
Mogłem ja to zrobić, ale przy okazji popraw można stworzyć nowy konstruktor w bazowym wyjątku, automatycznie wstawiający message "Failed...". W miarę możliwości podmienić w pozostałych plikach.